actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -609,7 +617,11 @@ def parse(name, encrypted_message)
         end
 
         def commit(options)
-          options[:value] = @encryptor.encrypt_and_sign(serialize(options[:value]))
+          if options[:expires].respond_to?(:from_now)
+            options[:expires] = options[:expires].from_now
+          end
```

Comment:
We can't share that implementation between the general cookie jar and the abstract cookie jar. So maybe we should put a `handle_options` before `commit` is called.

Something like:

```ruby
private
  delegate :handle_options, to: :@parent_jar
```

Then we'll eventually hit the `CookieJar` `handle_options`. Though it might be better to define a `preprocess_options` that specifically does the `from_now` handling.

Shows another fault in the cookies code, that we still have duplicated and split options processing, but I guess that's for later.